### PR TITLE
Sign in ml

### PR DESCRIPTION
**Pull request by mlimberg**
#### This PR includes the following:
* Ability to sign in
    * Sets active user in store with all compositions and sounds tied to that ID
* Error messages prompted if username and/or password do not match

#### Issue
* Modal does not disappear on blur
* Signing in then clicking logout shows sign in modal again

### CODE REVIEW

File: lib/containers/UserContainer.jsx [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
   postNewUser,
   logoutUser,
-  postNewUser
+  loginUser
 }
 
 const mapStateToProps = (state) => {
   return {
     user: state.ActiveUser.username,
-    error: state.ActiveUser.error || false
```
Comment (esayler) [inline lib/containers/UserContainer.jsx]: the || false could be refactored out?

File: lib/reducers/ActiveUser.js [hunk 2]
```diff
@@ -2,10 +2,11 @@
...
   switch(action.type) {
     case 'SET_USER':
-      const { id, email, username } = action.user
-      return { id, email, username }
+      return action.user
     case 'DUPLICATE':
-      return { error: action.error }
+      return { error: action.error, type: action.type }
+    case 'NOT_FOUND':
+      return { error: action.error, type: action.type }
```
Comment (esayler) [inline lib/reducers/ActiveUser.js]: Could be refactored to a single user error action, such as `USER_ERROR` that holds the error type and error message.

File: server.js [hunk 3]
```diff
@@ -444,3 +478,15 @@ app.get('*', function (request, response) {
...
+
+var something = 'Thank You'
+
+var say = (function(x) {
+    return function() { return x }
+})(something)
+
+something = 'Have a great day!';
+
+say();
```
Comment (esayler) [inline server.js]: remove this from the server.

**Comment by mlimberg** [comment]
@esayler @dylanavery720 @DanGrund 

#### Updated:
* Fixed all of Eric's comments
* Imported module to help hide modal on blur (react-onClickOutside)
* Added "X" close button to modal (button no longer toggles it with new blur feature)
* Added some minor styling to modal

**Comment by esayler** [comment]
Alright looks good 👍

**Pull request merged**